packages/twenty-server/src/engine/workspace-manager/workspace-health/services/database-structure.service.ts

```diff
@@ -160,14 +172,18 @@ export class DatabaseStructureService {
   getPostgresDataTypes(fieldMetadata: FieldMetadataEntity): string[] {
     const mainDataSource = this.typeORMService.getMainDataSource();
 
-    const normalizer = (type: FieldMetadataType, columnName: string) => {
+    const normalizer = (
+      type: FieldMetadataType,
+      isArray: boolean | undefined,
+      columnName: string,
+    ) => {
       const typeORMType = fieldMetadataTypeToColumnType(type);
 
       // Compute enum name to compare data type properly
       if (typeORMType === 'enum') {
-        const objectName = fieldMetadata.object?.nameSingular;
+        const objectName = computeObjectTargetTable(fieldMetadata.object);
```

Comment:
why do we need to pass isArray, can't we infer this from FieldMetadataType === 'MULTI_SELECT' directly?

Reply:
Because it can be based on a compositeType that contains directly an `isArray` property

Reply:
I see, I think we should refactor that once we do the FieldColumn Refactoring.
normalizer should only care about FieldColumn.ARRAY